File: mobile/app/src/main/java/com/smartsuschef/mobile/data/models/GlobalCalendarSignals.kt
package com.smartsuschef.mobile.data.models

data class GlobalCalendarSignals(
    val date: String,
    val isHoliday: Boolean,
    val holidayName: String,
    val isSchoolHoliday: Boolean,
    val rainMm: Double,
    val weatherDesc: String,
)

File: backend/SmartSusChef.Api/Models/GlobalCalendarSignals.cs
namespace SmartSusChef.Api.Models;

public class GlobalCalendarSignals
{
    public DateTime Date { get; set; }

    public bool IsHoliday { get; set; }
    
    public string HolidayName { get; set; } = string.Empty;

    public bool IsSchoolHoliday { get; set; }

    // ERD specifies decimal for RainMm to track precise rainfall data
    public decimal RainMm { get; set; }

    public string WeatherDesc { get; set; } = string.Empty;
}